inference-engine/tests/functional/plugin/gpu/dynamic_tests/gpu_dyn_batch_shape_tests.cpp

```diff
@@ -0,0 +1,103 @@
+    std::string,                                                       // Device name
+    std::map<std::string, std::string>                                 // Config
+>;
+
+class OVDynamicBatchShape_Tests : public WithParamInterface<OVDynamicBatchParams>,
+    virtual public ov::test::SubgraphBaseTest {
+public:
+    static std::string getTestCaseName(TestParamInfo<OVDynamicBatchParams> obj) {
+        std::vector<InputShape> inputShapes;
+        ElementType netPrecision;
+        std::string targetDevice;
+        std::map<std::string, std::string> configuration;
+        std::tie(inputShapes, netPrecision, targetDevice, configuration) = obj.param;
+
+        std::ostringstream result;
+        result << "IS=";
+        for (const auto& shape : inputShapes) {
+            result << CommonTestUtils::partialShape2str({ shape.first }) << "_";
+        }
+        result << "TS=";
+        for (const auto& shape : inputShapes) {
+            result << "(";
+            if (!shape.second.empty()) {
+                for (const auto& itr : shape.second) {
+                    result << CommonTestUtils::vec2str(itr);
+                }
+            }
+            result << ")_";
+        }
+        result << "netPRC=" << netPrecision << "_";
+        result << "targetDevice=" << targetDevice;
+        if (!configuration.empty()) {
+            for (auto& configItem : configuration) {
+                result << "configItem=" << configItem.first << "_" << configItem.second << "_";
+            }
+        }
+        return result.str();
+    }
+protected:
+protected:
+    void SetUp() override {
+        std::tie(inputShape, netPrecision, targetDevice, configuration) = this->GetParam();
+
+        init_input_shapes(inputShape);
+        //TODO: think how we can switch between several input topologies in the future
+        //  function = ngraph::builder::subgraph::makeSplitConvConcat(inputShape.front().first.get_min_shape(), netPrecision);
+        function = ngraph::builder::subgraph::makeSplitMultiConvConcat(inputShape.front().first.get_min_shape(), netPrecision);
+
+        //  make topology dynamic
+        std::map<std::string, ov::PartialShape> dynShape;
+        dynShape["input_tensor"] = inputShape.front().first;
+        function->reshape(dynShape);
+    }
+    std::shared_ptr<ov::runtime::Core> ie = ov::test::utils::PluginCache::get().core();
+    std::shared_ptr<ov::Model> src_func;
+    std::map<std::string, std::string> configuration;
+    std::vector<InputShape> inputShape;
+    ElementType netPrecision;
+};
+
+TEST_P(OVDynamicBatchShape_Tests, InferDynamicBatchBound) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+}
+
+namespace {
+const std::vector<std::map<std::string, std::string>> configs = {
+    {}
+};
+
+const std::vector<InputShape> inputShapes = {
+    { { {1, 19}, 4, 20, 20}, { {1, 4, 20, 20}, {7, 4, 20, 20}, {17, 4, 20, 20} }}
+};
+
+const std::vector<ElementType> netPrecisions = {
+        ElementType::f16,
```

Comment:
May be it makes sence to use 4-spaces tabulation here as in other parts of your code?

Reply:
Fixed